scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -740,34 +651,28 @@ private void removeJobRuntimeData(Session session, long jobId) {
 
     public void scheduleJobForRemoval(final JobId jobId, final long timeForRemoval, final boolean shouldRemoveFromDb) {
         // scheduleJobForRemoval
-        executeReadWriteTransaction(new SessionWork<Void>() {
-            @Override
-            public Void doInTransaction(Session session) {
-                session.getNamedQuery("setJobForRemoval")
-                       .setParameter("timeForRemoval", timeForRemoval)
-                       .setParameter("toBeRemoved", shouldRemoveFromDb)
-                       .setParameter("jobId", jobId.longValue())
-                       .executeUpdate();
-                return null;
-            }
+        executeReadWriteTransaction((SessionWork<Void>) session -> {
+            session.getNamedQuery("setJobForRemoval")
+                   .setParameter("timeForRemoval", timeForRemoval)
+                   .setParameter("toBeRemoved", shouldRemoveFromDb)
+                   .setParameter("jobId", jobId.longValue())
+                   .executeUpdate();
+            return null;
         });
     }
 
     public List<JobId> getJobsToRemove(final long time) {
-        List<JobId> jobIdsList = executeReadOnlyTransaction(new SessionWork<List<JobId>>() {
-            @Override
-            public List<JobId> doInTransaction(Session session) {
-                List<JobId> jobsToRemove = new ArrayList<JobId>();
-                Query query = session.createSQLQuery("select ID from JOB_DATA where " +
-                                                     "SCHEDULED_TIME_FOR_REMOVAL <> 0 and " +
-                                                     "SCHEDULED_TIME_FOR_REMOVAL < :timeLimit")
-                                     .setParameter("timeLimit", time);
-                Iterator jobIdIterator = query.list().iterator();
-                while (jobIdIterator.hasNext()) {
-                    jobsToRemove.add(JobIdImpl.makeJobId((jobIdIterator.next()).toString()));
-                }
-                return jobsToRemove;
-            }
+        List<JobId> jobIdsList = executeReadOnlyTransaction(session -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "jobIdsList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)